python/ray/tests/test_network_failure_e2e.py

```diff
@@ -0,0 +1,132 @@
+import sys
+import json
+
+from time import sleep
+import pytest
+from ray._private.test_utils import wait_for_condition
+from ray.tests.conftest_docker import *  # noqa
+from ray.tests.conftest_docker import gen_head_node, gen_worker_node
+
+
+SLEEP_TASK_SCRIPTS = """
+import ray
+ray.init(address="localhost:6379")
+@ray.remote(max_retries=-1)
+def f():
+    import time
+    time.sleep(10000)
+ray.get([f.remote() for _ in range(2)])
+"""
+
+head = gen_head_node(
+    {
+        "RAY_grpc_keepalive_time_ms": "1000",
+        "RAY_grpc_client_keepalive_time_ms": "1000",
+        "RAY_grpc_client_keepalive_timeout_ms": "1000",
+        "RAY_health_check_initial_delay_ms": "1000",
+        "RAY_health_check_period_ms": "1000",
+        "RAY_health_check_timeout_ms": "1000",
+        "RAY_health_check_failure_threshold": "2",
+    }
+)
+
+worker = gen_worker_node(
+    {
+        "RAY_grpc_keepalive_time_ms": "1000",
+        "RAY_grpc_client_keepalive_time_ms": "1000",
+        "RAY_grpc_client_keepalive_timeout_ms": "1000",
+        "RAY_health_check_initial_delay_ms": "1000",
+        "RAY_health_check_period_ms": "1000",
+        "RAY_health_check_timeout_ms": "1000",
+        "RAY_health_check_failure_threshold": "2",
+    }
+)
+
+
+def test_network_task_submit(head, worker, gcs_network):
+    network = gcs_network
+    # https://docker-py.readthedocs.io/en/stable/containers.html#docker.models.containers.Container.exec_run
+    head.exec_run(
+        cmd=f"python -c '{SLEEP_TASK_SCRIPTS}'",
+        detach=True,
+        environment=[
+            "RAY_grpc_client_keepalive_time_ms=1000",
+            "RAY_grpc_client_keepalive_timeout_ms=1000",
+        ],
+    )
+    sleep(3)
```

Comment:
no need this sleep since you have `wait_for_condition` later on.

Reply:
removed.

although we know that it will need to wait 3s to get a keepalive timeout